pkg/rest/compute/v2_1/flavor.go

```diff
@@ -286,6 +286,20 @@ func (svc *service) FlavorCreate(c *gin.Context) {
 		return
 	}
 
+	// It's a design limitation, while OpenStack Nova accepts
+	// flavors with same names. Our design use the flavor name as
+	// an uniq identifier.
+	flavor, err = clnt.Get(req.Flavor.Name)
+	if err != nil && !errors.IsNotFound(err) {
+		c.AbortWithError(http.StatusInternalServerError, err)
+		return
+	}
+
+	if flavor != nil {
+		c.AbortWithStatus(http.StatusConflict)
+		return
+	}
```

Comment:
Kubernetes should already enforce uniq names when we do the 'clnt.Create' call a few lines later below.

So IIUC, the key difference your change is making is that we report a StatusConflict, instead of StatusInternalServerError.

We can get the same effect by just adding a check "if errors.IsAlreadyExists(err)" after the Create call.

Reply:
> We can get the same effect by just adding a check "if errors.IsAlreadyExists(err)" after the Create call.

I will update the commit according your recommendation, I was not aware of that.